vm_test.go

```diff
@@ -2771,6 +2771,64 @@ export { x: 1 }
 		1)
 }
 
+func TestMethod(t *testing.T) {
+	expectRun(t, `
+	s := func(r)(n, m, o) {
+		f := "F"
+		g := "G"
+		h := n + m * o
+		return r.h
+	};
+	m := { f1: s, h: 22 }
+	out = m.f1(4, 5, 6)`,
+	nil, 22)
+
+	expectRun(t, `
+	m := { 
+		f1: func(r)(n) { 
+			if n > 0 { 
+				return r.res 
+			} else { 
+				return r.f1(n-1) 
+			}
+		}, 
+		res: 22
+	}
+	out = m.f1(12)`,
+	nil, 22)
+
+	expectRun(t, `
+	m := { f1: func(r)() { return r.c }}
+	n := copy(m)
+	n.c = 3
+	out = n.f1()`,
+	nil, 3)
+
+	expectRun(t, `
+	f1 := func(r)() { return r }
+	out = f1()`,
+	nil, tengo.UndefinedValue)
+
+	expectRun(t, `
+	m := { f1: func(r)() { return r.c }}
+	n := copy(m)
+	n.c = 3
+	out = n.f1()`,
+	nil, 3)
+
+	expectRun(t, `
+	m := { f1: func(r)() { return r == undefined}}
+	n := copy(m)
+	f := n.f1
+	out = f()`,
+	nil, true)
+
+	expectRun(t, `
+	m := { f1: func(r)() { return r == undefined}}
+	out = m.f1()`,
+	nil, false)
+}
```

Comment:
Add tests for array as well

Reply:
Thanks for pointing that out - it completely slipped my mind while writing the other cases.

I added a few that I think test the broadest categories of array operations. If there are any other unit tests needed, please let me know.